opensquirrel/circuit_builder.py

```diff
@@ -55,17 +52,17 @@ def __init__(
         self.register_manager = RegisterManager(qubit_register_size)
         self.ir = IR()
 
-    def __getattr__(self, attr: Any) -> Callable[..., Self]:
+    def __getattr__(self, attr: Any) -> Callable[..., CircuitBuilder]:
         if attr == "comment":
             return self._add_comment
 
         return partial(self._add_instruction, attr)
 
-    def _add_comment(self, comment_string: str) -> Self:
+    def _add_comment(self, comment_string: str) -> CircuitBuilder:
         self.ir.add_comment(Comment(comment_string))
         return self
 
-    def _add_instruction(self, attr: str, *args: Any) -> Self:
+    def _add_instruction(self, attr: str, *args: Any) -> CircuitBuilder:
```

Comment:
These functions returns `self` not `CircuitBuilder`.  This breaks type hinting when inheriting.